lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/persistence/DBOOAuthRefreshToken.java

```diff
@@ -0,0 +1,325 @@
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_OAUTH_REFRESH_TOKEN_HASH;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_OAUTH_REFRESH_TOKEN_ID;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_OAUTH_REFRESH_TOKEN_LAST_USED;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_OAUTH_REFRESH_TOKEN_MODIFIED_ON;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_OAUTH_REFRESH_TOKEN_NAME;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_OAUTH_REFRESH_TOKEN_PRINCIPAL_ID;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_OAUTH_REFRESH_TOKEN_SCOPES;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.DDL_OAUTH_REFRESH_TOKEN;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.TABLE_OAUTH_REFRESH_TOKEN;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.List;
+
+import org.sagebionetworks.repo.model.dbo.FieldColumn;
+import org.sagebionetworks.repo.model.dbo.MigratableDatabaseObject;
+import org.sagebionetworks.repo.model.dbo.TableMapping;
+import org.sagebionetworks.repo.model.dbo.migration.BasicMigratableTableTranslation;
+import org.sagebionetworks.repo.model.dbo.migration.MigratableTableTranslation;
+import org.sagebionetworks.repo.model.migration.MigrationType;
+
+/**
+ * Database object representing an OAuth 2.0 refresh token
+ */
+public class DBOOAuthRefreshToken implements MigratableDatabaseObject<DBOOAuthRefreshToken, DBOOAuthRefreshToken> {
+	private Long id;
+	private String tokenHash;
+	private String name;
+	private Long principalId;
+	private Long clientId;
+	private byte[] scopes;
+	private byte[] claims;
+	private Timestamp createdOn;
+	private Timestamp modifiedOn;
+	private Timestamp lastUsed;
+	private String etag;
+
+	private static FieldColumn[] FIELDS = new FieldColumn[] {
+		new FieldColumn("id", COL_OAUTH_REFRESH_TOKEN_ID, true).withIsBackupId(true),
+			new FieldColumn("tokenHash", COL_OAUTH_REFRESH_TOKEN_HASH),
+			new FieldColumn("name", COL_OAUTH_REFRESH_TOKEN_NAME),
+			new FieldColumn("principalId", COL_OAUTH_REFRESH_TOKEN_PRINCIPAL_ID),
+			new FieldColumn("clientId", COL_OAUTH_REFRESH_TOKEN_CLIENT_ID),
+			new FieldColumn("scopes", COL_OAUTH_REFRESH_TOKEN_SCOPES),
+			new FieldColumn("claims", COL_OAUTH_REFRESH_TOKEN_CLAIMS),
+			new FieldColumn("createdOn", COL_OAUTH_REFRESH_TOKEN_CREATED_ON),
+			new FieldColumn("modifiedOn", COL_OAUTH_REFRESH_TOKEN_MODIFIED_ON),
+			new FieldColumn("lastUsed", COL_OAUTH_REFRESH_TOKEN_LAST_USED),
+			new FieldColumn("etag", COL_OAUTH_REFRESH_TOKEN_ETAG).withIsEtag(true),
+		};
+
+	@Override
+	public TableMapping<DBOOAuthRefreshToken> getTableMapping() {
+		return new TableMapping<DBOOAuthRefreshToken>() {
+			// Map a result set to this object
```

Comment:
consider making this a static method to avoid the chance of referring to object members.